dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -763,9 +762,10 @@ public Workflow publish(@ApiParam(hidden = true) @Parameter(hidden = true, name
         @ApiParam(value = "Workflow id to publish/unpublish", required = true) @PathParam("workflowId") Long workflowId,
         @ApiParam(value = "PublishRequest to refresh the list of repos for a user", required = true) PublishRequest request) {
         Workflow workflow = workflowDAO.findById(workflowId);
-        checkEntry(workflow);
-
-        checkCanShareWorkflow(user, workflow);
+        checkNotNullEntry(workflow);
+        if (!isAdmin(user)) {
```

Comment:
It seems like this could be folded into `checkCanShare`, so that you don't have to do special checks (you have the same code in DockerRepoResource. But seems like you changed it on purpose, so...

Reply:
The `if (!isAdmin(user)) {` passages maintain admin privileges in a few cases where it didn't seem to matter one way or the other, to save some difficult IT modifications.